packages/odyssey-storybook/test/testcafe/framework/xpath-selector.ts

```diff
@@ -0,0 +1,19 @@
+import { Selector } from 'testcafe';
+
+const getElementsByXPath = Selector(xpath => {
+  const iterator = document.evaluate(xpath, document, null, XPathResult.UNORDERED_NODE_ITERATOR_TYPE, null);
+  const items = [];
+
+  let item = iterator.iterateNext();
+
+  while (item) {
+    items.push(item);
+    item = iterator.iterateNext();
+  }
+
+  return items;
+});
+
+export default function (xpath: string): Selector {
+  return Selector(getElementsByXPath(xpath));
```

Comment:
`testCafe` does not support finding the elements using XPath organically. I had two options:
1. Add another data selector such role or unique data-se
2. Add this helper to support finding elements by XPath.

Reply:
Why are we searching via XPath instead of using role or text selection?

I think this `xpath-selector` file can completely go. I didn't use any of these in Admin Sidenav.

Reply:
@KevinGhadyani-Okta were you able to write TestCafe selectors that felt more like testing library role or a11y attribute based selectors? That would be great to use here if possible. I glanced at the `Selector` TestCafe docs and did not see anything that seemed to math some of the implicit role selection that testing library provides although we could build our own by looking at their source if necessary.

All that said, I would love to keep this simple. The storybook is not performance sensitive and adding a few extra `data-testid="foo"` attributes to our stories themselves to make selection here easier is not a bad outcome. I would prefer that over including complicated or custom xpath selector logic

Reply:
Yeah. `@testing-library/testcafe` has all the necessary selectors so you can use `queryByRole('button')`.

Reply:
I am not opposed to using the Testing Library (it was my initial go-to actually). However, I wasn't able to get it to work with the existing tags and attributes. I will ask for your help @KevinGhadyani-Okta to make it work with queryByRole (or other selectors)